src/components/NewCommentForm/NewCommentForm.tsx

```diff
@@ -0,0 +1,105 @@
+import './NewCommentForm.scss';
+import React, { FormEvent, useState } from 'react';
+import { Comment } from '../../types';
+
+type Props = {
+  commentsList: Comment[] | undefined;
+  setCommentsList: (arg:Comment[]) => void;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+                                                  commentsList, setCommentsList,
+                                                }) => {
+  const [yourname, setYourname] = useState('');
+  const [youremail, setYouremail] = useState('');
+  const [yourcomment, setYourcomment] = useState('');
+  const [error, setError] = useState(false);
+
+  const addComment = (newComment: Comment) => {
+    if (commentsList) {
+      commentsList.unshift(newComment);
+      const copy = [...commentsList];
+
+      setCommentsList(copy);
+    }
+  };
+
+  const submitHandler = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (yourname && youremail && yourcomment && commentsList) {
+      const newComment = {
+        id: commentsList.length + 1,
+        postId: commentsList[0].postId,
+        name: yourname,
+        email: youremail,
+        body: yourcomment,
+      };
+
+      addComment(newComment);
+      setError(false);
+      setYourname('');
+      setYouremail('');
+      setYourcomment('');
```

Comment:
Since we are not making any form validation, we can use uncontrolled components, so that we can avoid react from handling the form states.


Another option is to use react-hook-form, which is a library to handle forms in react, it uses uncontrolled components by default but let us create them in a controlled way, isolating the component, achieving a better performance on the way.